src/test/mir-opt/const_debuginfo.main.ConstDebugInfo.diff

```diff
@@ -79,13 +79,16 @@
                                            // + span: $DIR/const_debuginfo.rs:14:13: 14:28
                                            // + literal: Const { ty: &str, val: Value(Slice { data: Allocation { bytes: [104, 101, 108, 108, 111, 44, 32, 119, 111, 114, 108, 100, 33], relocations: Relocations(SortedMap { data: [] }), init_mask: InitMask { blocks: [8191], len: Size { raw: 13 } }, align: Align { pow2: 0 }, mutability: Not, extra: () }, start: 0, end: 13 }) }
           StorageLive(_10);                // scope 5 at $DIR/const_debuginfo.rs:16:9: 16:10
+          Deinit(_10);                     // scope 5 at $DIR/const_debuginfo.rs:16:13: 16:34
```

Comment:
Hmm... I wonder if we can/should optimize this pattern. If the local which we deinit wasn't touched since it went live, we don't need to deinit.

Reply:
We could definitely optimize this, but I don't think it practically matters too much. We don't codgen `deinit` statements right now anyway, so I doubt this matters a lot for perf. If we ever get enough optimizations set up to make this easy it would be a good thing to have, but it does not seem worth going out of our way for now

Reply:
Makes sense, thanks